huxley/core/admin/assignment.py

```diff
@@ -39,30 +41,69 @@ def load(self, request):
         '''Loads new Assignments.'''
         assignments = request.FILES
         reader = csv.reader(assignments['csv'])
+        successful_rows = []
+        failed_rows = []
 
         def get_model(model, name, cache):
             if not name in cache:
                 cache[name] = model.objects.get(name=name)
             return cache[name]
 
-        def generate_assigments(reader):
+        def get_bad_columns(row):
+            bad_columns = set()
+            if not Committee.objects.filter(name=row[1]).exists():
+                bad_columns.add(1)
+            if not Country.objects.filter(name=row[2]).exists():
+                bad_columns.add(2)
+            if not School.objects.filter(name=row[0]).exists():
+                bad_columns.add(0)
+            return bad_columns
+
+        def create_assignments(reader):
             committees = {}
             countries = {}
             schools = {}
+            assigned = set()
+
             for row in reader:
                 if (row[0]=='School' and row[1]=='Committee' and row[2]=='Country'):
                     continue # skip the first row if it is a header
-                committee = get_model(Committee, row[1], committees)
-                country = get_model(Country, row[2], countries)
-                school = get_model(School, row[0], schools)
+                
+                while len(row) < 3:
+                    row.append("") # extends the row to have the minimum proper num of columns
+
+                bad_columns = get_bad_columns(row)
+                already_assigned = (row[1], row[2]) in assigned
+                
                 if len(row) < 4:
                     rejected = False # allow for the rejected field to be null
                 else:
                     rejected = (row[3].lower() == 'true') # use the provided value if admin provides it
-                yield (committee.id, country.id, school.id, rejected)
```

Comment:
We should most definitely return a generator from this function. It takes up a lot of space otherwise